.github/workflows/ci-cd.yml

```diff
@@ -115,13 +115,13 @@ jobs:
       uses: actions/cache@v4.0.0
       id: cache
       with:
-        key: llhttp-${{ hashFiles('vendor/llhttp/package.json', 'vendor/llhttp/src/**/*') }}
+        key: llhttp-${{ hashFiles('vendor/llhttp/package*.json', 'vendor/llhttp/src/**/*') }}
```

Comment:
BTW, I updated this to include the lock file, but consider revamping this key to something safer.  There are other files that could affect the build, such as the `Makefile`, `bin/generator.ts`, and maybe others if I dug deeper.  Consider just getting the commit SHA from the checkout and using that as the key.

Reply:
To be honest, this really only needs to know if the submodule has changed, right? So, if that information can be obtained, it could actually be simplified a lot.

Reply:
Yes, exactly.  I would just get the commit SHA of the submodule folder and use that as key.  IMO, there's some other streamlining of this job that could be done, so probably best for a different PR (e.g. I think only one checkout is needed, and no need to both cache and upload an artifact of the build).

Reply:
I'll merge this one for now, feel free to add additional improvements.